src/pkg/cli/composeUp.go

```diff
@@ -75,26 +75,31 @@ func ComposeUp(ctx context.Context, loader client.Loader, c client.FabricClient,
 	}
 
 	deployRequest := &defangv1.DeployRequest{Mode: mode, Project: project.Name, Compose: bytes, DelegateDomain: delegateDomain.Zone}
+
+	delegation, err := p.PrepareDomainDelegation(ctx, client.PrepareDomainDelegationRequest{
+		DelegateDomain: delegateDomain.Zone,
+		Preview:        upload == compose.UploadModePreview,
+		Project:        project.Name,
+	})
+	if err != nil {
+		return nil, project, err
+	} else if delegation != nil {
+		deployRequest.DelegationSetId = delegation.DelegationSetId
+	}
```

Comment:
Looking at the code, isn't `delegation` guaranteed to not be `nil` at this point?

```suggestion
	}
	deployRequest.DelegationSetId = delegation.DelegationSetId
```

I think we can avoid the similar check on line 97 as well.

Reply:
no, DO and Playground return `nil, nil` so we still can get nil.